src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/WorkspaceProject.cs

```diff
@@ -144,18 +144,17 @@ public async Task SetBuildSystemPropertiesAsync(IReadOnlyDictionary<string, stri
                 case "TargetPath": _project.OutputFilePath = GetFullyQualifiedPath(valueOrNull); break;
                 case "TargetRefPath": _project.OutputRefFilePath = GetFullyQualifiedPath(valueOrNull); break;
                 case "TargetFrameworkIdentifier": _targetFrameworkManager.UpdateIdentifierForProject(_project.Id, valueOrNull); break;
+                case "IntermediateAssembly": _project.CompilationOutputAssemblyFilePath = GetFullyQualifiedPath(valueOrNull); break;
             }
         }
 
-        // Workaround for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1830960
-        _project.CompilationOutputAssemblyFilePath = _project.OutputFilePath;
-
         string? GetFullyQualifiedPath(string? propertyValue)
         {
-            Contract.ThrowIfNull(_project.FilePath, "We don't have a project path at this point.");
+            var projectPath = Path.GetDirectoryName(_project.FilePath);
```

Comment:
Since you're adding a call to `Path.GetDirectoryName`, it could throw other exceptions like `PathTooLongException` do we care to guard against those?

Reply:
Good question. I would imagine if the machine didn't have long paths enabled, but a project file exists that is longer than that path, then MSBuild would have already thrown (these values all come from MSBuild) but I don't honestly know what to expect here. @jasonmalinowski any ideas?

Easy enough to make this defensive just in case I suppose.

Reply:
Yeah I don't see a reason to guard here -- we'd be likely to crash a lot earlier, and there's nothing much this code can do other than just try to silently march along which would just break other stuff. Best just to let the exceptions happen.